internal/plugin/data_plane_server.go

```diff
@@ -34,7 +36,9 @@ type (
 		instances    []*instances.Instance
 		configEvents map[string][]*instances.ConfigurationStatus
 		messagePipe  bus.MessagePipeInterface
-		server       net.Listener
+		server       *http.Server
+		ctx          context.Context
```

Comment:
Is there another way to do this where we don't store the `context.Context` inside the struct?
https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx: